node_modules.py

```diff
@@ -301,8 +301,8 @@ def write_rpm_sources(fh, args):
         if args.source_offset is not None:
             i += 1
 
-def process_packagelock_file(js):
-    if not "lockfileVersion" in js or js["lockfileVersion"] != 2:
+def process_packagelock_file(js, ignore_lockfile_version):
+    if not "lockfileVersion" in js or (not ignore_lockfile_version and js["lockfileVersion"] != 2):
```

Comment:
if we are to ignore the lockfile version, maybe just add `if not ignore_lockfile_version:` otherwise if `lockfileVersion` is missing and it should be ignored, it's not ignored.

But my main comment here is that `npm` seems to require lockfileVersion: 2 to fetch all the dependencies (aka, bundled dependencies). Without all the dependencies available, npm will try to ask the registry for them, even when not installing, and then it fails as there is no network connection. Do you have a different experience here? Or does your OBS have network access at build time?